pkg/cmd/server/command.go

```diff
@@ -0,0 +1,151 @@
+	flag.Var(&cfg.EtcdAddr, "etcd", "The address of the etcd server (host, host:port, or URL). If specified, no built-in etcd will be started.")
+	flag.Var(&cfg.KubernetesAddr, "kubernetes", "The address of the Kubernetes server (host, host:port, or URL). If specified, no Kubernetes components will be started.")
+	flag.Var(&cfg.KubernetesPublicAddr, "public-kubernetes", "The Kubernetes server address for use by public clients, if different. (host, host:port, or URL). Defaults to same as --kubernetes.")
+	flag.Var(&cfg.PortalNet, "portal-net", "A CIDR notation IP range from which to assign portal IPs. This must not overlap with any IP ranges assigned to nodes for pods.")
+
+	flag.StringVar(&cfg.ImageTemplate.Format, "images", cfg.ImageTemplate.Format, "When fetching images used by the cluster for important components, use this format on both master and nodes. The latest release will be used by default.")
+	flag.BoolVar(&cfg.ImageTemplate.Latest, "latest-images", cfg.ImageTemplate.Latest, "If true, attempt to use the latest images for the cluster instead of the latest release.")
+
+	flag.StringVar(&cfg.VolumeDir, "volume-dir", "openshift.local.volumes", "The volume storage directory.")
+	flag.StringVar(&cfg.EtcdDir, "etcd-dir", "openshift.local.etcd", "The etcd data directory.")
+	flag.StringVar(&cfg.CertDir, "cert-dir", "openshift.local.certificates", "The certificate data directory.")
+
+	flag.StringVar(&cfg.Hostname, "hostname", cfg.Hostname, "The hostname to identify this node with the master.")
+	flag.Var(&cfg.NodeList, "nodes", "The hostnames of each node. This currently must be specified up front. Comma delimited list")
+	flag.Var(&cfg.CORSAllowedOrigins, "cors-allowed-origins", "List of allowed origins for CORS, comma separated.  An allowed origin can be a regular expression to support subdomain matching.  CORS is enabled for localhost, 127.0.0.1, and the asset server by default.")
+
+	cfg.ClientConfig = cmdutil.DefaultClientConfig(flag)
+
+	cfg.Docker.InstallFlags(flag)
+
+	return cmd, cfg
+}
+
+const startMaster = "master"
+const startNode = "node"
+
+func (cfg Config) Validate(args []string) error {
+	switch len(args) {
+	case 1:
+		switch args[0] {
+		case startMaster: // allowed case
+		case startNode: // allowed case
+		default:
+			return errors.New("You may start an OpenShift all-in-one server with no arguments, or pass 'master' or 'node' to run in that role.")
+		}
+	case 0:
+		// do nothing, this starts an all in one
+
+	default:
+		return errors.New("You may start an OpenShift all-in-one server with no arguments, or pass 'master' or 'node' to run in that role.")
+	}
+
+	return nil
+}
+
+// Complete takes the args and fills in information for the start config
+func (cfg *Config) Complete(args []string) {
+	cfg.ExplicitStartMaster = (len(args) == 1) && (args[0] == startMaster)
+	cfg.ExplicitStartNode = (len(args) == 1) && (args[0] == startNode)
```

Comment:
I still think it would be cleaner for the switch statement in start.go do something like:
```
case cfg.StartMaster && cfg.StartNode:
  ... all in one
case cfg.StartMaster:
  ... master
case cfg.StartNode:
  ... node
```
but I don't care that much